docs/modules/ROOT/pages/references/architecture/espejote-in-cluster-templating-controller.adoc

```diff
@@ -0,0 +1,502 @@
+= Espejote: An in-cluster templating controller
+
+== Problem Statement
+
+Not every cluster configuration is fully static and can be managed through GitOps.
+
+Some configurations depend on the cluster state or might want to be managed by upstream, by end-users, or by third-party products directly.
+
+We got an extensive array of tools to manage these configurations, from bash "reconcilers," to cron jobs, to policy engines to fully fledged custom controllers.
+Those tools are all specialized and don't cover all use-cases.
+
+A maintainable, fully instrumented, and extensible controller that can manage arbitrary configurations in a cluster is missing.
+We decided to try our hand at creating one.
+
+== High Level Goals
+
+* We can manage arbitrary configurations in a cluster
+** The configuration may depend on external state
```

Comment:
maybe clarify external state here? is the state external, but in-cluster? or external behind an arbitrary URL?

Reply:
clarified as `state managed by other controllers inside the cluster`